src/devices/GoPiGo3/Models/GoPiGoInfo.cs

```diff
@@ -39,7 +40,7 @@ public class GoPiGoInfo
         /// <returns>Returns an int array of the hardware version</returns>
         public int[] GetHardwareVersion()
         {
-            return GetVersionsFromString(HardwareVersion);
+            return GetVersionsFromString(HardwareVersion.ToString());
```

Comment:
I think this is not needed now but if still want to have it I'd change GetVersionsFromString to GetVersionAsArray and make it simply access the fields of the Version class

Reply:
I can remove those functions. They have a limited utility with the Version class